### PR TITLE
add range endpoints for leaf, block, payload and header

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: src/availability.rs [hunk 1]
```diff
@@ -171,6 +171,26 @@ where
...
+                state
+                    .get_leaf_range(from..until)
+                    .await
+                    .enumerate()
+                    .then(|(index, fetch)| async move {
+                        fetch.with_timeout(timeout).await.context(FetchLeafSnafu {
+                            resource: (index + from).to_string(),
+                        })
+                    })
+                    .map(|result| result.map(LeafQueryData::from))
```
Comment (jbearer) [inline src/availability.rs:188]: Is there an error if we don't have this `LeafQueryData::from`? I thought `get_leaf_range` is already returning a stream of `LeafQueryData`.

File: src/availability.rs [hunk 2]
```diff
@@ -243,6 +283,26 @@ where
...
+                state
+                    .get_block_range(from..until)
+                    .await
+                    .enumerate()
+                    .then(|(index, fetch)| async move {
+                        fetch.with_timeout(timeout).await.context(FetchBlockSnafu {
+                            resource: (index + from).to_string(),
+                        })
+                    })
+                    .map(|result| result.map(BlockQueryData::from))
```
Comment (jbearer) [inline src/availability.rs:300]: same question here

File: src/availability.rs [hunk 3]
```diff
@@ -275,6 +335,26 @@ where
...
+                state
+                    .get_payload_range(from..until)
+                    .await
+                    .enumerate()
+                    .then(|(index, fetch)| async move {
+                        fetch.with_timeout(timeout).await.context(FetchBlockSnafu {
+                            resource: (index + from).to_string(),
+                        })
+                    })
+                    .map(|result| result.map(PayloadQueryData::from))
```
Comment (jbearer) [inline src/availability.rs:352]: and here

File: src/availability.rs (hunk 1 above)
Comment (imabdulbasit) [reply src/availability.rs:188]: No error.

**Comment by imabdulbasit** [comment]
force pushed the main branch mistakenly. creating PR again

**Pull request closed without merging**